rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/MtuWatcher.java

```diff
@@ -0,0 +1,45 @@
+package com.polidea.rxandroidble.internal.connection;
+
+
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.inject.Inject;
+import javax.inject.Named;
+import rx.Completable;
+import rx.CompletableSubscriber;
+import rx.Subscription;
+import rx.functions.Action1;
+
+@ConnectionScope
+class MtuWatcher extends Completable implements MtuProvider {
+
+    private final AtomicInteger currentMtuAtomicInteger;
+
+    @Inject
+    MtuWatcher(
+            final RxBleGattCallback rxBleGattCallback,
+            final AtomicInteger atomicInteger,
+            @Named(ConnectionComponent.NamedInts.GATT_MTU_MINIMUM) final int initialValue
+    ) {
+        super(new OnSubscribe() {
+            @Override
+            public void call(CompletableSubscriber completableSubscriber) {
+                Subscription mtuSubscription = rxBleGattCallback.getOnMtuChanged()
+                        .retry()
```

Comment:
What is the scenario for `retry()` is required at all?

Reply:
When the `BluetoothGattCallback.onMtuChanged()` would be called with `status != GATT_SUCCESS`